packages/node/src/localEvaluation.ts

```diff
@@ -0,0 +1,184 @@
+    const resultsString = evaluation.evaluate(
+      JSON.stringify(flagConfigs),
+      JSON.stringify(user),
+    );
+    this.debug('[Experiment] evaluate - result:', resultsString);
+    // Parse variant results
+    const results: EvaluationResult = JSON.parse(resultsString);
+    const variants: Variants = {};
+    Object.keys(results).forEach((key) => {
+      variants[key] = {
+        value: results[key].variant.key,
+        payload: results[key].variant.payload,
+      };
+    });
+    return variants;
+  }
+
+  /**
+   * Start polling for flag configurations.
+   *
+   * You must call this function to begin polling for flag config updates.
+   * The promise returned by this function is resolved when the initial call
+   * to fetch the flag configuration completes.
+   *
+   * Calling this function while the poller is already running will stop and
+   * restart the poller.
+   */
+  public async start(): Promise<void> {
+    this.stop();
+    this.debug('[Experiment] poller - start');
+    this.flagConfigPoller = setInterval(async () => {
+      await this.updateFlagConfigs();
+    }, this.config.flagConfigPollingInterval);
+    if (!this.flagConfigPromise) {
+      this.flagConfigPromise = this.updateFlagConfigs({
+        attempts: 5,
+        min: 1,
+        max: 1,
+        scalar: 1,
+      });
+      await this.flagConfigPromise;
+    }
+  }
+
+  /**
+   * Stop polling for flag configurations.
+   *
+   * Calling this function while the poller is not running will do nothing.
+   */
+  public stop(): void {
+    if (this.flagConfigPoller) {
+      this.debug('[Experiment] poller - stop');
+      clearTimeout(this.flagConfigPoller);
+      this.flagConfigPoller = undefined;
+    }
+  }
+
+  private async fetchFlagConfigs(): Promise<Record<string, string>> {
+    const flagConfigs = await this.doFlagConfigs();
+    return this.parseFlagConfigs(flagConfigs);
+  }
+
+  private async getFlagConfigs(flagKeys?: string[]): Promise<string[]> {
+    if (this.flagConfigPromise) {
+      this.debug('[Experiment] waiting for flag configs');
+      await this.flagConfigPromise;
+    }
+    return Object.values(this.flagConfigCache.get(flagKeys));
+  }
+
+  private async updateFlagConfigs(
+    backoffPolicy?: BackoffPolicy,
+  ): Promise<void> {
+    this.debug('[Experiment] updating flag configs');
+    return await doWithBackoff<void>(async () => {
+      const flagConfigs = await this.fetchFlagConfigs();
+      this.flagConfigCache.clear();
+      this.flagConfigCache.put(flagConfigs);
+    }, backoffPolicy);
+  }
+
+  private async doFlagConfigs(): Promise<string> {
+    const endpoint = `${this.config.serverUrl}/sdk/rules?d=fdsa`;
+    const headers = {
+      Authorization: `Api-Key ${this.apiKey}`,
+    };
+    this.debug('[Experiment] Get flag configs');
+    const response = await this.httpClient.request(
+      endpoint,
+      'GET',
+      headers,
+      null,
```

Comment:
comment or initialize a variable with null and then pass it; can't read looking at code here what this param